functions/New-DbaDbTransfer.ps1

```diff
@@ -0,0 +1,246 @@
+            'Users',
+            'Roles',
+            'PartitionSchemes',
+            'PartitionFunctions',
+            'XmlSchemaCollections',
+            'SqlAssemblies',
+            'UserDefinedAggregates',
+            'UserDefinedTypes',
+            'Schemas',
+            'Synonyms',
+            'Sequences',
+            'DatabaseTriggers',
+            'DatabaseScopedCredentials',
+            'ExternalFileFormats',
+            'ExternalDataSources',
+            'Logins',
+            'ExternalLibraries'
+        )]
+        [string[]]$CopyAll,
+
+        [switch]$SchemaOnly,
+
+        [switch]$DataOnly,
+
+        [switch]$EnableException
+    )
+    begin {
+        $objectCollection = New-Object System.Collections.ArrayList
+    }
+    process {
+        foreach ($object in $InputObject) {
+            if (-not $object) {
+                Stop-Function -Message "Object is empty"
+                return
+            }
+            $objectCollection.Add($object) | Out-Null
+        }
+
+    }
+    end {
+        $sourceDb = Get-DbaDatabase -SqlInstance $SqlInstance -SqlCredential $SqlCredential -Database $Database -EnableException:$EnableException
+        if (-not $sourceDb) {
+            Stop-Function -Message "Database not found"
+            return
+        } elseif ($sourceDb.Count -gt 1) {
+            Stop-Function -Message "More than one database found with the parameters provided"
+            return
+        }
+        # Create transfer object and define properties based on parameters
+        $transfer = New-Object Microsoft.SqlServer.Management.Smo.Transfer($sourceDb)
+        foreach ($object in $objectCollection) {
+            $transfer.ObjectList.Add($object) | Out-Null
+        }
+        $transfer.BatchSize = $BatchSize
+        $transfer.BulkCopyTimeOut = $BulkCopyTimeOut
+        $transfer.CopyAllObjects = $CopyAllObjects
+        foreach ($copyType in $CopyAll) {
+            $transfer."CopyAll$copyType" = $true
+        }
+        if ($ScriptingOption) { $transfer.Options = $ScriptingOption }
+
+        # Add destination connection parameters
+        if ($DestinationSqlInstance.IsConnectionString) {
+            $connString = $DestinationSqlInstance.InputObject
+        } elseif ($DestinationSqlInstance.Type -eq 'RegisteredServer' -and $DestinationSqlInstance.InputObject.ConnectionString) {
+            $connString = $DestinationSqlInstance.InputObject.ConnectionString
+        } elseif ($DestinationSqlInstance.Type -eq 'Server' -and $DestinationSqlInstance.InputObject.ConnectionContext.ConnectionString) {
+            $connString = $DestinationSqlInstance.InputObject.ConnectionContext.ConnectionString
+        } else {
+            $transfer.DestinationServer = $DestinationSqlInstance.InputObject
+            $transfer.DestinationLoginSecure = $true
+        }
+        if ($connString) {
+            $connStringBuilder = New-Object System.Data.SqlClient.SqlConnectionStringBuilder $connString
```

Comment:
Any reason to not use `New-DbaConnectionStringBuilder` to lesson the amount of code in this command?

Reply:
The builder here is to parse the connection string, not build it. No reason to increase the amount of code being called when I can get away with creating an object via a one-liner.

Reply:
Is that what `New-DbaConnectionString` can be used for then?

Reply:
A single line of code (222)  could be replaced with New-DbaConnectionString, yes. But that's hardly worth it.

Reply:
I'm referring to lines 211 through 220 that lead up to using the `$connString` variable. I don't have a test system setup but is `$DestinationSqlInstance | New-DbaConnectionString` not doable to perform the same thing?

Reply:
From what I can see, that function only accepts a string as a parameter and has no other logic built into it other than replacing/adding connection string parameters.